app/models/setting/general.rb

```diff
@@ -26,6 +26,7 @@ def self.default_settings
       self.set('lab_features', N_("Whether or not to show a menu to access experimental lab features (requires reload of page)"), false, N_('Show Experimental Labs')),
       self.set("append_domain_name_for_hosts", N_("Foreman will append domain names when new hosts are provisioned"), true, N_("Append domain names to the host")),
       self.set('outofsync_interval', N_("Duration in minutes after servers are classed as out of sync."), 30, N_('Out of sync interval')),
+      self.set('instance_id', N_("Foreman instance ID, uniquely identifies this Foreman instance."), 'uuid', N_('Foreman UUID'), Foreman.uuid),
```

Comment:
Is it actually desirable for users to change it?

Reply:
Yes, in case of LB setup, where 2 instances should act as one. Also that allows to easily opt-out in case people don't want it to be set.

Reply:
I'd argue, that in a LB setup the instances share the same database. E.g. the instance id is automatically the same for all lb targets. I also think that we should not allow an opt-out of the instance ID. As a developer, I want to rely on having the instance ID being always available. If we send the ID is part of some tracking callback and should provide a way to opt out.
If we decide to make the instance ID available in core, people should only be allowed to change it via a rake task that asks for confirmation imho. This should not be an arbitrary text field.

Reply:
You're right about the DB so instances will share the UUID anyway. Given the feedback we got here, we slowed down in using this for any kind of tracking (rss user-agent header) and rather will be more careful, allowing people to opt-out by setting any value. If we'll go ahead and will start using this in other usecases too, I think it would be good to introduce read-only settings. This would be handy for other settings. I suppose the question arised - why settings if they can't change it.

The answer would be, we already have this. Settings stored in yaml files are displayed in UI but can be modified. That is useful to easily figure out the value, perhaps even get it via API. The benefit in this case would be, it's stored in DB instead of a file, which is easier to handle in LB setup. If we decide in future to lock it, then I'd agree it would be also handy to have resetting or changing rake task.

Is the current approach acceptable now? I spoke with @ekohl and we agreed to improve commit message which addresses his concerns. Hopefully this would lead to consensus.

Reply:
Makes sense. If we don't use it for tracking, I'm good. Let's provide a (tiny) proper API to consume this. So we can change the implementation in the future.

Reply:
Just to be clear, you mean some developer API like ˋForeman.instance_id(=)`? I like the idea. I may even add Jail object/template macro for it (in separate PR ). @ShimShtein, could you please add it to the PR? I think we'd be ready after it.